src/artm/regularizer/smooth_ptdw.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright 2014, Additive Regularization of Topic Models.
+
+// Author: Murat Apishev (great-mel@yandex.ru)
+
+#include <vector>
+
+#include "glog/logging.h"
+#include "artm/core/protobuf_helpers.h"
+#include "artm/utility/blas.h"
+
+#include "artm/regularizer/smooth_ptdw.h"
+
+namespace artm {
+namespace regularizer {
+
+void SmoothPtdwAgent::Apply(int item_index, int inner_iter, ::artm::utility::DenseMatrix<float>* ptdw) const {
+  int local_token_size = ptdw->no_rows();
+  int topics_count = ptdw->no_columns();
+  if (config_.type() == SmoothPtdwConfig_Type_MovingAverage) {
+    // 1. evaluate wich tokens are background
+    double treshold = config_.treshold();
+    std::vector<bool> is_background(local_token_size, false);
+    int count_background = 0;
+    for (int i = 0; i < local_token_size; ++i) {
+      const float* local_ptdw_ptr = &(*ptdw)(i, 0);  // NOLINT
+      double sum_background = 0.0;
+      for (int k = 0; k < topics_count; ++k) {
+        char b = 'b';
```

Comment:
We definitely should address this issue, but perhaps later... Ideally we should describe which topics are background in parameters of this regularizer. However none of other regularizers has this concept. Also, what if user did not configure background topics? Does it mean that ``smooth_ptdw`` regularizer is not applicable?

Maybe an alternative solution would be to define a list of all background words as a dictionary, and ask user to build a separate topic model in order to create this dictionary.